test/unittests/types_test.cpp

```diff
@@ -0,0 +1,46 @@
+// Fizzy: A fast WebAssembly interpreter
+// Copyright 2019-2020 The Fizzy Authors.
+// SPDX-License-Identifier: Apache-2.0
+
+#include "types.hpp"
+#include <gtest/gtest.h>
+
+using namespace fizzy;
+using namespace testing;
+
+TEST(types, functype)
+{
+    FuncType functype_v = {};
+    FuncType functype_v_v = {};
+    FuncType functype_i = {{ValType::i32}, {}};
+    FuncType functype_ii = {{ValType::i32, ValType::i32}, {}};
+    FuncType functype_I = {{ValType::i64}, {}};
+    FuncType functype_i_i = {{ValType::i32}, {ValType::i32}};
+    FuncType functype_i_ii = {{ValType::i32}, {ValType::i32, ValType::i32}};
+    FuncType functype_ii_i = {{ValType::i32, ValType::i32}, {ValType::i32}};
```

Comment:
@gumb0 if you want something nicer than these tests (e.g. some loop generating cases) then please take it over.

Reply:
no, it looks great :-)